tests/modules/images/webp-uploads/webp-uploads-test.php

```diff
@@ -809,6 +809,28 @@ function( $transforms ) {
 		}
 	}
 
+	/**
+	 * Update source attributes when webp is edited.
+	 *
+	 * @test
+	 */
+	public function it_should_validate_source_attribute_update_when_webp_edited() {
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+
+		$editor = new WP_Image_Edit( $attachment_id );
+		$editor->crop( 10, 10, 0, 0 )->save();
+		$this->assertTrue( $editor->success() );
+
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+
+		$this->assertArrayHasKey( 'sources', $metadata );
+		$this->assertArrayHasKey( 'sizes', $metadata );
```

Comment:
This look good it would be great to include the custom assertion to verify the sources like

```
		$this->assertImageHasSource( $attachment_id, 'image/webp' );
		$this->assertImageNotHasSource( $attachment_id, 'image/jpeg' );
```

You can use the same assertions for the size names as well.

```
https://github.com/WordPress/performance/blob/5790b8027e9a4216de2709304b251dbba413bcab/tests/modules/images/webp-uploads/webp-uploads-test.php#L807-L808
```

Also, be aware that because of the dimensions of your edit not all subsizes would be affected so you can use a higher crop that affects all subsizes or you can perform a different edit like a flip.

Once assertion that would be wort checking is making sure the actual values from the sources are edited files you can use the following assertion:

```php
$this->assertRegExp( '/e\d{13}/', $file_name );
```

Due edited images have a `e{13_DIGIT_NUMBER}` to ensure the values on the meta are actual edited versions of the file and not the original image.

Reply:
@mitogh Thanks a lot for these pointers. I have updated the code to include all the newer assertions as well.

Reply:
Thanks, great work on getting this one ready.